webapp/src/ts/services/db-sync.service.ts

```diff
@@ -166,31 +168,76 @@ export class DBSyncService {
       telemetryEntry.recordSuccess(result);
     } catch (err) {
       telemetryEntry.recordFailure(err, this.knownOnlineState);
-      console.error(`Error replicating from remote server`, err);
+      console.error('Error replicating from remote server', err);
       return err;
     }
   }
 
+  private getCurrentSeq(): Promise<number> {
+    return this.dbService.get().info().then(info => info.update_seq);
+  }
 
-  private getCurrentSeq() {
-    return this.dbService.get().info().then(info => info.update_seq + '');
+  private getLastReplicatedSeq(): number {
+    return Number(window.localStorage.getItem(LAST_REPLICATED_SEQ_KEY)) || 0;
   }
 
-  private getLastReplicatedSeq() {
-    return window.localStorage.getItem(LAST_REPLICATED_SEQ_KEY);
+  private setLastReplicatedSeq(sequence: number) {
+    if (!sequence) {
+      return;
+    }
+
+    const lastReplicatedSeq = this.getLastReplicatedSeq();
+    if (lastReplicatedSeq > sequence) {
+      return;
+    }
+
+    window.localStorage.setItem(LAST_REPLICATED_SEQ_KEY, sequence.toString());
   }
 
   private getLastReplicationDate() {
     return window.localStorage.getItem(LAST_REPLICATED_DATE_KEY);
   }
 
+  private async makeBidirectionalReplication(replicateFromServer: boolean, successiveSyncs = 0) {
```

Comment:
Thanks for breaking up the functions!

The naming on this one is a little complicated to me... I think it's because "bidirectional replication" is just a long way of saying "sync" which is what we're using throughout this service. Because we already have a `syncMedic` function it makes the naming of this difficult but feel free to change the naming of that function so how this and that work together is clear.

Reply:
I struggle with this naming, I renamed them like this:
syncMedic --> startSyncMedic 🫤 (open to suggestions)
makeBidirectionalReplication --> syncMedic